protocol-units/settlement/mcr/contracts/test/token/MOVETokenV2.t.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import "forge-std/Test.sol";
+import "../../src/token/MOVETokenDev.sol";
+import {ProxyAdmin} from "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import {TransparentUpgradeableProxy} from "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {IAccessControl} from "@openzeppelin/contracts/access/IAccessControl.sol";
+
+contract MOVETokenDevTest is Test {
+    MOVETokenDev public token;
+    ProxyAdmin public admin;
+    string public moveSignature = "initialize(address)";
+    address public multisig = 0x00db70A9e12537495C359581b7b3Bc3a69379A00;
+    bytes32 public MINTER_ROLE;
+
+    function setUp() public {
+        MOVETokenDev moveTokenImplementation = new MOVETokenDev();
+
+        // Deploy proxies
+        TransparentUpgradeableProxy moveProxy = new TransparentUpgradeableProxy(
+            address(moveTokenImplementation), address(multisig), abi.encodeWithSignature(moveSignature, multisig)
+        );
+        token = MOVETokenDev(address(moveProxy));
+        MINTER_ROLE = token.MINTER_ROLE();
+    }
+
+    function testCannotInitializeTwice() public {
+        vm.startPrank(multisig);
+        // Initialize the contract
+        vm.expectRevert();
+        token.initialize(multisig);
+        vm.stopPrank();
+    }
+
+    function testGrants() public {
+        // Check the token details
+        assertEq(token.hasRole(MINTER_ROLE, multisig), true);
+    }
+
+    function testMint() public {
+        vm.startPrank(multisig);
+        uint256 intialBalance = token.balanceOf(address(0x1337));
+        // Mint tokens
+        token.mint(address(0x1337), 100);
+
+        // Check the token details
+        assertEq(token.balanceOf(address(0x1337)), intialBalance + 100);
+        vm.stopPrank();
+    }
+
+    function testRevokeMinterRole() public {
+        vm.startPrank(multisig);
+        assertEq(token.hasRole(MINTER_ROLE, multisig), true);
+
+        token.mint(address(0x1337), 100);
+        // Revoke minter role
+        token.revokeMinterRole(multisig);
+
+        // Check the token details
+        assertEq(token.hasRole(MINTER_ROLE, multisig), false);
+
+        vm.expectRevert(
+            abi.encodeWithSelector(IAccessControl.AccessControlUnauthorizedAccount.selector, multisig, MINTER_ROLE)
+        );
+        token.mint(address(0x1337), 100);
+        vm.stopPrank();
+    }
+
+    function testGrantRevokeMinterAdminRole() public {
```

Comment:
this test is a bit convoluted. can you add a description here as comment what you want to test here actually.